src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+    }
+
+    public boolean canConnectRedstoneDefault(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side,
+            MicroblockContainer partContainer)
+    {
+        return super.canConnectRedstone(state, world, pos, side);
+    }
+
+    @Override
+    public final int getWeakPower(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side)
+    {
+        if (side == null) { return 0; }
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container == null) { return getWeakPowerDefault(state, world, pos, side, null); }
+        return Math.max(container.getPartContainer().getWeakSignal(side), getWeakPowerDefault(state, world, pos, side, container));
+    }
+
+    public int getWeakPowerDefault(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side, MicroblockContainer partContainer)
+    {
+        return super.getWeakPower(state, world, pos, side);
+    }
+
+    @Override
+    public final int getStrongPower(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side)
+    {
+        if (side == null) { return 0; }
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container == null) { return getStrongPowerDefault(state, world, pos, side, null); }
+        return Math.max(container.getPartContainer().getStrongSignal(side), getStrongPowerDefault(state, world, pos, side, container));
+    }
+
+    public int getStrongPowerDefault(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side,
+            MicroblockContainer partContainer)
+    {
+        return super.getStrongPower(state, world, pos, side);
+    }
+
+    @Override
+    public final boolean isSideSolid(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container == null) { return false; }
+        return container.getPartContainer().isSideSolid(side) || isSideSolidDefault(state, world, pos, side);
+    }
+
+    public boolean isSideSolidDefault(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side)
+    {
+        return super.isSideSolid(state, world, pos, side);
+    }
+
+    @Override
+    public final boolean canPlaceTorchOnTop(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container == null) { return false; }
+        return container.getPartContainer().canPlaceTorchOnTop() || canPlaceTorchOnTopDefault(state, world, pos);
```

Comment:
All of these early returns you do like this need to be switched to if (!=null)'s 1) it's cleaner and 2) it makes code easier to understand and 3) it makes future expansion easier without need to refactor these two lines.

Not to mention all you single line ifs need to be converted into two lined one minus the braces.